components/StoryListItem/StoryListItem.tsx

```diff
@@ -1,23 +1,78 @@
 import { ClockIcon, GlobeAltIcon } from "../icons";
-import { StoryWithTopics, Topic } from "../../lib/db";
+import { StoryWithListEntities, Topic } from "../../lib/db";
+import useCreateReaction from "../../hooks/useCreateReaction";
+import useFetchReactionOptions from "../../hooks/useFetchReactionOptions";
+import useDeleteReaction from "../../hooks/useDeleteReaction";
+import useUser from "../../hooks/useUser";
 
 interface StoryListItemProps {
-  story: StoryWithTopics;
+  story: StoryWithListEntities;
 }
 
 const StoryListItem: React.FC<StoryListItemProps> = ({ story }) => {
+  const reactionOptions = useFetchReactionOptions();
+
   const sourceUrl = new URL(story.sourceUrl);
   sourceUrl.searchParams.append("utm_source", "rodinia_report");
   sourceUrl.searchParams.append("ref", "rodinia_report");
 
   const createdAt = new Intl.DateTimeFormat().format(new Date(story.createdAt));
+  const user = useUser();
+  const createReactionMutation = useCreateReaction();
+  const deleteReactionMutation = useDeleteReaction();
+  const isLoading =
+    createReactionMutation.isLoading || deleteReactionMutation.isLoading;
 
   const storyData = {
     createdAt,
     sourceUrl: sourceUrl.toString(),
     sourceHostname: new URL(story.sourceUrl).hostname,
   };
 
+  type ReactionOptionData = {
+    [key: number]: {
+      id: number;
+      name: string;
+      count: number;
+      storyReactionId?: number;
+    };
+  };
+
+  let reactionOptionData: ReactionOptionData = {};
+
+  if (!reactionOptions.isLoading && !user.isLoading) {
```

Comment:
I think ideally the API could give us a data structure similar to what we'll need for the UI. This way we won't need to be doing all this work in the UI.